.travis.yml

```diff
@@ -6,26 +6,32 @@ services:
 matrix:
   fast_finish: true
   include:
-    - php: 7.1
+    - php: 7.3
```

Comment:
Please keep PHP 7.1. We are still supporting this version of PHP

Reply:
Is it a deliberate decision to continue support of outdated PHP versions? PHP 7.1 and 7.2 are [end-of-life](https://www.php.net/eol.php). They are no longer receiving security updates and installations should be upgraded to newer PHP versions as soon as possible.

On top of that, PHP 7.1 and 7.2 are [not supported by the latest PHPUnit](https://phpunit.de/supported-versions.html), PHPUnit 9. I could use a wider version constraint for PHPUnit (something like `^7.5 || ^8.0 || ^9.0`) to allow tests on these PHP versions, but it might add some additional complexity. Of course, if support for these PHP versions really is required this is the only way to go.

Reply:
Interesting thanks! @curry684 what's your opinion?

Reply:
I know the Symfony PhpUnit Bridge could solve that issue but I am not a fan of the way it works.

Reply:
Now that PHPUnit 8.5 allow PHP 8, I believe it should be possible to run on phpunit `^7.5 || ^8.5` from php 7.1 to 8, with the same tests. Technically as long as the bundle supports symfony 4 IMO php 7.1 should be supported too.

Reply:
In general for libraries like this I'm in favor of following PHP support life cycles, so drop a PHP version once it's clearly end of life. So not the next day, but by all means with the next big update like the one we're discussing here.

However, it should most of all match the project requirements in my opinion, so *if we declare to support it we should test for it*. Hence the Travis PHP versions should match the minimum PHP version in `composer.json`. If you remove 7.1 from the tests, change `composer.json` to 7.2+.

Reply:
thanks for your insights @Seldaek and @curry684 ! Much appreciated.

I am in favor of dropping PHP 7.1 support and keep PHP 7.2. And thus it should be tested against this version in the CI.

Reply:
> Hence the Travis PHP versions should match the minimum PHP version in `composer.json`. If you remove 7.1 from the tests, change `composer.json` to 7.2+.

I agree, I had already changed the PHP version requirement in `composer.json` to `"^7.3 || ^8.0"`. If the conclusion of this thread is to continue supporting PHP 7.2 and/or 7.1, I'll make sure that they are re-introduced in `composer.json` as well as in the CI configuration.

As per @Seldaek's comment, [the addition of PHP 8 support to PHPUnit 8.5](https://github.com/sebastianbergmann/phpunit/blob/master/ChangeLog-8.5.md#8512---2020-11-30) has simplified this a bit, leaving us with three options:

1. Keep this PR as it is (support PHP 7.3, 7.4 and 8.0 using PHPUnit 9)
2. Support PHP 7.2, 7.3, 7.4 and 8.0 using PHPUnit 8.5
3. Support PHP 7.1, 7.2, 7.3, 7.4 and 8.0 using PHPUnit 7.5|8.5

In case of option 2 or 3 I might have to revert some of the changes to the test code but that shouldn't be a big issue.
I can imagine how option 3 might introduce some deprecation catch-22's (where a method has been deprecated in 8.5 but no alternative exists yet in 7.5) but I haven't checked that yet so it may not be a problem at all.

Perhaps it's good to keep in mind that this decision will come back with each new release of PHP. When PHP 8.1 will be released at the end of this year, it will not be supported by PHPUnit 7 or 8, but only by PHPUnit 9 (or even 10, I'm not sure). At some point you will be forced to choose between supporting the latest PHP version and older, EOL versions.

Reply:
`2.` looks like a good solution to me. Thanks @nicwortel :pray: